pbt-101.py

```diff
@@ -310,6 +359,16 @@ def test_record_json_roundtrip(record):
 #       back to a new record, and finally to json again.  Are these strings
 #       equal?  Would this have the same problem as test_record_json_roundtrip?
 
+# I wasn't sure how to approach this - I get the concept of caching but
+# am not sure how to implement it in Python. Did a bit of reading and it looks
+# like `pickle` might be an option, but there were a few blog posts that
+# recommended _against_ using pickle.
```

Comment:
You could also use a simple `dict` in memory, or store it as JSON-encoded strings in this case!  For production uses you would presumably have some cache service, but the details don't really matter here.  See e.g. the standard library `functools.lru_cache` for an easy implementation example!

Reply:
This is super useful Zac, as always. Þakka þér fyrir.
_Caught up with a few other things currently, but will loop back to more Python practice when back in CBR_